src/browser_action/vitals.js

```diff
@@ -260,12 +260,10 @@
 
     else if (metric.name == 'CLS' && metric.entries.length) {
       metric.entries.forEach(entry => {
-        console.log('Layout shift:', {
-          'score': Math.round(entry.value * 10000) / 10000,
-          'elements': entry.sources.map(source => {
-            return source.node;
-          })
-        });
+        console.log('Layout shift - score: ', Math.round(entry.value * 10000) / 10000);
+        entry.sources.map(source => {
```

Comment:
It makes the control flow linear instead of inverting it through the callback. And generally why make a function, have a closure, etc for `forEach` when you can use a regular loop.

FWIW had the code already used `forEach` I wouldn't normally have mentioned that preference in a code review. Some people like `forEach` and most of the time it doesn't matter :)

Reply:
Works for me. Found this too: https://stackoverflow.com/questions/43031988/javascript-efficiency-for-vs-foreach

Change it in https://github.com/GoogleChrome/web-vitals-extension/pull/122/commits/fa9a637d368207e89234117097ee6a1d2138d9cd

Reply:
> Found this too: https://stackoverflow.com/questions/43031988/javascript-efficiency-for-vs-foreach

To add to that answer, the performance story is a little more complicated because a `forEach` _can_ be inlined and essentially become a for loop, and the `for...of` isn't the same thing as a classic for loop, because it's using an iterator under the hood, which can end up doing various things perf-wise depending on what the rest of the code is doing.

But in many cases (like this one), the loop is so small the overhead doesn't matter either way, so to me comes down more to what's more straightforward when reading the code.